File: js/app_bar/app_bar.js
define('app_bar/app_bar', ['permission', 'search_bar/search_bar', 'utils'],
    function() {
  return angular.module('AppBarModule', ['PermissionModule',
     'SearchBarModule', 'UtilsModule', 'ServicesModule'])
    .directive('appBar', function(perm, utils, rpc) {
      return {
        scope: {
          admining: '@',
          user: '='
        },
        link: function(scope) {
          scope.weekDayLabels = utils.weekDayLabels;
          scope.formatTime = utils.formatTime;
          scope.isAdmin = function() {
            return perm.isAdmin();
          };
          scope.isSysAdmin = function() {
            return perm.isSysAdmin();
          };
          scope.isDistrictInspector = () => perm.isDistrictInspector();
          scope.isCountryInspector = () => perm.isCountryInspector();
          scope.isYearLeader = function() {
            return perm.isYearLeader();
          };
          scope.isOrderAdmin = function() {
            return perm.isOrderAdmin();
          };
          scope.isTeacher = function() {
            return perm.isTeacher();
          };
          scope.canReadOrders = function() {
            return perm.canReadOrders();
          };
          scope.showAssignments = function() {
            return perm.isSysAdmin() || perm.isYearLeader();
          };
          scope.isIn = function(page) {
            return location.pathname.endsWith(page);
          };
          scope.selectedClassId = null;
          scope.changeClass = function() {
            if (!scope.selectedClassId) return;
            utils.redirect('php/su.php?class_id={0}'.format(scope.selectedClassId));
          };
          rpc.get_user_classes().then(function(response) {
            scope.classes = response.data;
            scope.selectedClassId = (scope.user || perm.user).classInfo.id;
          });
        },
        templateUrl : 'js/app_bar/app_bar.html?tag=202512122357'
      };
    });
});

File: v2/lib/components/app_bar/app_bar.dart
import 'dart:html';

import 'package:angular/angular.dart';
import 'package:angular_components/material_input/material_input.dart';
import 'package:v2/model/user.dart';
import 'package:v2/services/user_service.dart';

@Component(
  selector: 'app-bar',
  directives: [MaterialInputComponent, NgIf],
  templateUrl: 'app_bar.html',
  styleUrls: ['app_bar.css'],
)
class AppBarComponent {
  final UserService _userService;

  AppBarComponent(this._userService);

  User get user => _userService.user;

  bool isIn(String page) => window.location.pathname.endsWith(page);
  bool isAdmin() => false;
  bool isSysAdmin() => false;
  bool isDistrictInspector() => false;
  bool isYearLeader() => false;

  bool isOrderAdmin() => false;
  bool isTeacher() => false;
  bool canReadOrders() => false;
  bool showAssignments() => false;

  bool get admining => false;
}
